### PR TITLE
precompile: if a single package is requested, show output live

### PR DESCRIPTION
**Pull request by IanButterworth**
Fixes https://github.com/JuliaLang/Pkg.jl/issues/3487

@vtjnash something like this, perhaps?

Screenshot 2023-08-11 at 9 04 01 PM
Screenshot 2023-08-11 at 9 04 32 PM

### CODE REVIEW

**Comment by IanButterworth** [comment]
An alternative would be for code loading precompilation to parallel precomp all of the depdendencies but leave off the requested package, then exit and leave the old serial precompilation to finish the main package like it used to. 
Though it would look a bit like a bug if it did that, I think.

File: src/API.jl [hunk 1]
```diff
@@ -1277,10 +1278,21 @@ function precompile(ctx::Context, pkgs::Vector{PackageSpec}; internal_call::Bool
     taskwaiting = Set{Base.PkgId}()
     pkgspidlocked = Dict{Base.PkgId,String}()
 
-    function monitor_stderr(pkg, iob)
+    function monitor_stderr(pkg, iob; single_requested_pkg=false)
         try
+            liveprinting = false
             while isopen(iob)
                 str = readline(iob)
```
Comment (vtjnash) [inline src/API.jl:1285]:
```suggestion
                str = readline(iob; keep=true)
```
(will give better performance)

**Comment by IanButterworth** [comment]
Updated to
Screenshot 2023-08-14 at 10 09 16 PM

Note that now if you call `precompile` in a package project, it will follow this behavior too.

**Comment by IanButterworth** [comment]
> I like the concept of having output appear as each task finishes for each process. 

It's an interesting idea, but would be a bigger change with some tricky UX because of the way packages disappear when done. This PR is a bit more manageable in UX because it only happens to the last singular package being precompiled.

**Comment by IanButterworth** [comment]
Let's get this on master to get it tried out

**Pull request merged**